src/pubtools/sign/signers/msgsigner.py

```diff
@@ -467,12 +468,19 @@ def msg_container_sign(signing_key=None, task_id=None, config="", digest=None, r
 )
 @click.option("--task-id", required=True, help="Task id identifier (usually pub task-id)")
 @click.option("--config", default=CONFIG_PATHS[0], help="path to the config file")
+@click.option("--raw", default=False, is_flag=True, help="Print raw output instead of json")
 @click.argument("inputs", nargs=-1)
-def msg_clear_sign_main(inputs, signing_key=None, task_id=None, config=None):
+def msg_clear_sign_main(inputs, signing_key=None, task_id=None, config=None, raw=None):
     """Entry point method for clearsign operation."""
-    click.echo(
-        json.dumps(msg_clear_sign(inputs, signing_key=signing_key, task_id=task_id, config=config))
-    )
+    ret = msg_clear_sign(inputs, signing_key=signing_key, task_id=task_id, config=config)
+    if not raw:
+        click.echo(json.dumps(ret))
+    if ret["signer_result"]["status"] == "error":
```

Comment:
Aren't you missing an "else" here? If raw is not selected, I think both raw and non-raw output gets printed.

Reply:
yeah, vim-failure